modules/eks/cluster/main.tf

```diff
@@ -121,7 +121,7 @@ module "eks_cluster" {
   public_access_cidrs          = var.public_access_cidrs
   subnet_ids                   = var.cluster_private_subnets_only ? local.private_subnet_ids : concat(local.private_subnet_ids, local.public_subnet_ids)
   vpc_id                       = local.vpc_id
-  addons                       = var.addons
+  addons                       = local.addons
   addons_depends_on            = var.addons_depends_on ? [module.region_node_group] : null
```

Comment:
Enhance `addons_depends_on` to depend on `aws_iam_role_policy_attachment`s regardless of `local.addons`. (It safe to always depend on them, even when disabled, just omit the index.)